src/main/webapp/WEB-INF/pages/productDetails.jsp

```diff
@@ -1,10 +1,19 @@
 <%@ page contentType="text/html;charset=UTF-8" %>
 <%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
 <%@ taglib prefix="tags" tagdir="/WEB-INF/tags" %>
 
 
 <jsp:useBean id="product" type="com.es.phoneshop.model.product.Product" scope="request"/>
+<jsp:useBean id="recentlyViewedProducts" type="java.util.List" scope="request"/>
```

Comment:
i hold only products id in session and load products from DAO by id. I think if we were using database instead of local storage it would be possible to display wrong info about products (if for example our product price changes in database it might not be changed in session (if session holds objects), but if we have id we will always have valid product from our DAO.

Reply:
ah ok, then I got your point
didn't notice that you stored only id-s
thanks for the explanation, I will resolve all related comments